### PR TITLE
chain: Make db handle each transaction instead of event

### PR DESCRIPTION
**Pull request by sorawit**

### CODE REVIEW

File: chain/app/db_app.go [hunk 1]
```diff
@@ -48,14 +51,24 @@ func (app *dbBandApp) DeliverTx(req abci.RequestDeliverTx) (res abci.ResponseDel
...
-	if res.IsOK() && lastProcessHeight+1 == app.DeliverContext.BlockHeight() {
-		for _, event := range res.Events {
-			kvMap := make(map[string]string)
-			for _, kv := range event.Attributes {
-				kvMap[string(kv.GetKey())] = string(kv.GetValue())
-			}
-			app.dbBand.HandleEvent(event.Type, kvMap)
-		}
+	if lastProcessHeight+1 == app.DeliverContext.BlockHeight() {
+		return res
```
Comment (sorawit) [inline chain/app/db_app.go:55]: This is not correct, as @taobun pointed out.

File: chain/db/db.go [hunk 2]
```diff
@@ -50,18 +54,37 @@ func (b *BandDB) RollBack() {
 	b.tx = nil
 }
 
-func (b *BandDB) HandleEvent(eventName string, attributes map[string]string) {
-	switch eventName {
+func (b *BandDB) HandleTransaction(tx auth.StdTx, txHash []byte, logs sdk.ABCIMessageLogs) {
```
Comment (sorawit) [inline chain/db/db.go:57]: We may also need other information like block time, block height. Might even be better to just pass Context here?

**Comment by sorawit** [comment]
@taobun Please take this forward

File: chain/db/db.go (hunk 2 above)
Comment (taobun) [reply chain/db/db.go:57]: We need info about that Send context here or save context in bandDB state?

**Comment by sorawit** [comment]
Close in favor of #957

**Pull request closed without merging**